R/methods-epi_archive.R

```diff
@@ -1000,3 +1000,79 @@ epix_truncate_versions_after.epi_archive <- function(x, max_version) {
   return((x$clone()$truncate_versions_after(max_version)))
   # ^ second set of parens drops invisibility
 }
+
+#' Print and summary functions for an `epi_archive` object.
+#'
+#' @param x The `epi_archive` object.
+#' @param class Boolean; whether to print the class label header
+#' @param methods Boolean; whether to print all available methods of
+#'   the archive
+#' @param ... Additional arguments passed to methods.
+#'
+#' @method print epi_archive
+#' @export
+print.epi_archive <- function(x, class = TRUE, methods = TRUE, ...) {
+  # --- Copied from R6 ---
+  if (class) cat("An `epi_archive` object, with metadata:\n")
+  cat(sprintf("* %-9s = %s\n", "geo_type", x$geo_type))
+  cat(sprintf("* %-9s = %s\n", "time_type", x$time_type))
+  if (!is.null(x$additional_metadata)) {
+    sapply(x$additional_metadata, function(m) {
+      cat(sprintf("* %-9s = %s\n", names(m), m))
+    })
+  }
+  cat("----------\n")
+  if (length(x$DT$time_value) == 0 || all(is.na(x$DT$time_value))) {
+    min_time <- max_time <- NA
+  } else {
+    min_time <- Min(x$DT$time_value)
+    max_time <- Max(x$DT$time_value)
+  }
+  cat(sprintf("* %-14s = %s\n", "min time value", min_time))
+  cat(sprintf("* %-14s = %s\n", "max time value", max_time))
+  cat(sprintf(
+    "* %-14s = %s\n", "first version with update",
+    min(x$DT$version)
+  ))
+  cat(sprintf(
+    "* %-14s = %s\n", "last version with update",
+    max(x$DT$version)
+  ))
```

Comment:
Do we need min/max version statistics?

Reply:
Min version could be misleading.... archive should represent the FULL version history up to some point.  Just because the first report in that history was the min version doesn't mean that this doesn't/shouldn't represent the full history.

Max version, yes, probably.  Helps users spot when their archive might need a refresh.